mirascope/prompts.py

```diff
@@ -78,16 +93,12 @@ def template(cls) -> str:
 
     def __str__(self) -> str:
         """Returns the docstring prompt template formatted with template variables."""
-        template = self.template()
-        template_vars = [
-            var for _, var, _, _ in Formatter().parse(template) if var is not None
-        ]
-        return template.format(**{var: getattr(self, var) for var in template_vars})
+        return _format_template(self, self.template())
```

Comment:
add `self: Prompt` like below?

Reply:
Unnecessary. Inside of the `Prompt` class it already knows that `self` is a `Prompt` (but for the `messages_fn` below it doesn't know what self is because it's a function external to the class and then set as an attribute).